crates/modelardb_server/src/parser.rs

```diff
@@ -875,58 +876,61 @@ mod tests {
     }
 
     #[test]
-    fn test_tokenize_and_parse_create_model_table_with_lowercase_keyword_as_table_name() {
-        for keyword in ALL_KEYWORDS {
-            // END-EXEC cannot be parsed by the SQL parser because of the hyphen, and is therefore
-            // skipped in this test.
-            if keyword == &"END-EXEC" {
-                continue;
-            }
-            let keyword_lowercase = keyword.to_lowercase();
-            let error = tokenize_and_parse_sql(
-                format!(
-                    "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
-                    keyword_lowercase
-                )
-                .as_str(),
-            );
+    fn test_tokenize_and_parse_create_table_with_lowercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_lowercase,
+            "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+        )
+    }
 
-            assert!(error.is_err());
+    #[test]
+    fn test_tokenize_and_parse_create_table_with_uppercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_uppercase,
+            "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+        )
+    }
 
-            assert_eq!(
-                error.unwrap_err(),
-                ParserError::ParserError(format!(
-                    "Reserved keyword '{}' cannot be used as a table name.",
-                    keyword_lowercase
-                ))
-            );
-        }
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_with_lowercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_lowercase,
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
     }
 
     #[test]
     fn test_tokenize_and_parse_create_model_table_with_uppercase_keyword_as_table_name() {
+        parse_and_assert_if_keyword_is_restricted_in_table_name(
+            str::to_uppercase,
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
+    }
+
+    fn parse_and_assert_if_keyword_is_restricted_in_table_name(
```

Comment:
Minor but should probably be `parse_and_assert_that_keywords_are_restricted_in_table_name` instead.

Reply:
Fixed.